src/apps/bluebit/aurora/bluebit.farm.contract-position-fetcher.ts

```diff
@@ -0,0 +1,75 @@
+import { Inject } from '@nestjs/common';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { RewardRateUnit } from '~app-toolkit/helpers/master-chef/master-chef.contract-position-helper';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { ContractPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+
+import { BLUEBIT_DEFINITION } from '../bluebit.definition';
+import { Bluebit, BluebitContractFactory } from '../contracts';
+
+const appId = BLUEBIT_DEFINITION.id;
+const groupId = BLUEBIT_DEFINITION.groups.farm.id;
+const network = Network.AURORA_MAINNET;
+
+@Register.ContractPositionFetcher({ appId, groupId, network })
+export class AuroraBluebitFarmContractPositionFetcher implements PositionFetcher<ContractPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(BluebitContractFactory) private readonly bluebitContractFactory: BluebitContractFactory,
+  ) {}
+
+  async getPositions() {
+    return this.appToolkit.helpers.masterChefContractPositionHelper.getContractPositions<Bluebit>({
+      address: '0x947dD92990343aE1D6Cbe2102ea84eF73Bc5790E',
+      appId,
+      groupId,
+      network,
+      dependencies: [
+        {
+          appId,
+          groupIds: [BLUEBIT_DEFINITION.groups.vault.id],
+          network,
+        },
+      ],
+      resolveContract: opts => this.bluebitContractFactory.bluebit(opts),
+      resolvePoolLength: async ({ multicall, contract }) => multicall.wrap(contract).poolLength(),
+      resolvePoolIndexIsValid: async ({ poolIndex, multicall }) => {
+        switch (poolIndex) {
+          case 2:
+          case 3:
+          case 13:
+          case 14:
+            return false;
+          default:
+            return true;
+        }
+      },
+      resolveDepositTokenAddress: async ({ poolIndex, contract, multicall }) => {
+        const pool = await multicall.wrap(contract).pools(poolIndex);
+        const vault = this.bluebitContractFactory.vault({ address: pool.vault, network: network });
+        const token = await vault.swapPair();
+        return token.toLowerCase();
+      },
+      resolveTotalValueLocked: async ({ multicall, contract, poolIndex }) => {
+        const pool = await multicall.wrap(contract).pools(poolIndex);
+        const vault = this.bluebitContractFactory.vault({ address: pool.vault, network: network });
+        const totalSupply = await vault.totalSupply();
+        return totalSupply;
+      },
```

Comment:
Just a heads up, you'll need to sync your fork with the upstream. The function here has been renamed to `resolveLiquidity` instead of `resolveTotalValueLocked`.

Reply:
fixed